src/rules.ts

```diff
@@ -391,14 +391,17 @@ export class RuleChain extends LogicRule {
   ): Promise<IRuleResult> {
     const result = await this.evaluate(parent, args, ctx, info, options)
 
-    if (result.some(res => res !== true)) {
+    if (result.slice(-1).some(res => res !== true)) {
       const customError = result.find(res => res instanceof Error)
       return customError || false
     } else {
       return true
     }
   }
 
+  static isBreak = (res: any) => res !== true
```

Comment:
hmm, I think we could just add this to the new rule and leave `RuleChain` unchanged.